daemon/network.go

```diff
@@ -302,6 +303,12 @@ func (daemon *Daemon) createNetwork(create types.NetworkCreateRequest, id string
 		driver = c.Config().Daemon.DefaultDriver
 	}
 
+	isSwarmModeManager := daemon.clusterProvider != nil && daemon.clusterProvider.IsManager()
+	withClusterStore := daemon.configStore.ClusterStore != "" // classic
+	if driver == "overlay" && !isSwarmModeManager && !withClusterStore {
```

Comment:
Thanks @AkihiroSuda, I agree with this change as it will remove the very confusing previous message.
But I think we should not base it on the driver name (`overlay`), because we support network plugins which can be global scope, therefore working with previous cluster store config and with new swarm.

What do you think about have libnetwork return a well known error in case the datastore is not present 
https://github.com/docker/libnetwork/blob/master/store.go#L228
so that you can recognize it at L331 here and return the new error instead ?
You can define the error in https://github.com/docker/libnetwork/blob/master/error.go